Marlin/src/lcd/extui/lib/mks_ui/tft_lvgl_configuration.cpp

```diff
@@ -556,174 +513,23 @@ unsigned int getTickDiff(unsigned int curTick, unsigned int lastTick) {
-  if ((*y) > XPT2046_Y_MIN) (*y) -= XPT2046_Y_MIN; else (*y) = 0;
-  (*x) = uint32_t(uint32_t(*x) * XPT2046_HOR_RES) / (XPT2046_X_MAX - XPT2046_X_MIN);
-  (*y) = uint32_t(uint32_t(*y) * XPT2046_VER_RES) / (XPT2046_Y_MAX - XPT2046_Y_MIN);
-  #if XPT2046_X_INV
-    (*x) = XPT2046_HOR_RES - (*x);
-  #endif
-  #if XPT2046_Y_INV
-    (*y) = XPT2046_VER_RES - (*y);
-  #endif
-}
-
-#define times 4
-#define CHX   0x90
-#define CHY   0xD0
-
-int SPI2_ReadWrite2Bytes(void) {
-  #define SPI_READ_WRITE_BYTE(B) TERN(TFT_LVGL_UI_SPI, SPI_TFT.spi_read_write_byte, W25QXX.spi_flash_read_write_byte)(B)
-  const uint16_t t1 = SPI_READ_WRITE_BYTE(0xFF),
-                 t2 = SPI_READ_WRITE_BYTE(0xFF);
-  return (((t1 << 8) | t2) >> 3) & 0x0FFF;
-}
+static bool get_point(int16_t *x, int16_t *y) {
+  bool is_touched = touch.getRawPoint(x, y);
 
-uint16_t x_addata[times], y_addata[times];
-void XPT2046_Rd_Addata(uint16_t *X_Addata, uint16_t *Y_Addata) {
-  uint16_t i, j, k;
-
-  TERN(TFT_LVGL_UI_SPI, SPI_TFT.spi_init, W25QXX.init)(SPI_SPEED_6);
-
-  for (i = 0; i < times; i++) {
-    #if ENABLED(TFT_LVGL_UI_SPI)
-      OUT_WRITE(TOUCH_CS_PIN, LOW);
-      SPI_TFT.spi_read_write_byte(CHX);
-      y_addata[i] = SPI2_ReadWrite2Bytes();
-      WRITE(TOUCH_CS_PIN, HIGH);
-
-      OUT_WRITE(TOUCH_CS_PIN, LOW);
-      SPI_TFT.spi_read_write_byte(CHY);
-      x_addata[i] = SPI2_ReadWrite2Bytes();
-      WRITE(TOUCH_CS_PIN, HIGH);
-    #else // #if HAS_TOUCH_XPT2046
-      OUT_WRITE(TOUCH_CS_PIN, LOW);
-      W25QXX.spi_flash_read_write_byte(CHX);
-      y_addata[i] = SPI2_ReadWrite2Bytes();
-      WRITE(TOUCH_CS_PIN, HIGH);
-
-      OUT_WRITE(TOUCH_CS_PIN, LOW);
-      W25QXX.spi_flash_read_write_byte(CHY);
-      x_addata[i] = SPI2_ReadWrite2Bytes();
-      WRITE(TOUCH_CS_PIN, HIGH);
-    #endif
-
-  }
-  TERN(TFT_LVGL_UI_SPI,,W25QXX.init(SPI_QUARTER_SPEED));
-
-  for (i = 0; i < times; i++)
-    for (j = i + 1; j < times; j++)
-      if (x_addata[j] > x_addata[i]) {
-        k = x_addata[j];
-        x_addata[j] = x_addata[i];
-        x_addata[i] = k;
-      }
-  if (x_addata[times / 2 - 1] - x_addata[times / 2] > 50) {
-    *X_Addata = *Y_Addata = 0;
-    return;
-  }
-
-  *X_Addata = (x_addata[times / 2 - 1] + x_addata[times / 2]) / 2;
-
-  for (i = 0; i < times; i++)
-    for (j = i + 1; j < times; j++)
-      if (y_addata[j] > y_addata[i]) {
-        k = y_addata[j];
-        y_addata[j] = y_addata[i];
-        y_addata[i] = k;
-      }
-
-  if (y_addata[times / 2 - 1] - y_addata[times / 2] > 50) {
-    *X_Addata = *Y_Addata = 0;
-    return;
+  if (is_touched) {
+    *x = int16_t((int32_t(*x) * XPT2046_X_CALIBRATION) >> 16) + XPT2046_X_OFFSET;
+    *y = int16_t((int32_t(*y) * XPT2046_Y_CALIBRATION) >> 16) + XPT2046_Y_OFFSET;
   }
 
-  *Y_Addata = (y_addata[times / 2 - 1] + y_addata[times / 2]) / 2;
-}
-
-#define ADC_VALID_OFFSET  10
+  #if ENABLED(GRAPHICAL_TFT_ROTATE_180)
+    x = 480 - x;
+    y = 320 - y;
```

Comment:
The 480 and 320 here should probably be `LCD_FULL_PIXEL_WIDTH`/`HEIGHT` and in general any mysterious numbers used more than a few times should be given named defines.

Reply:
Fixed